cluster/node-pools/worker-karpenter/provisioners.yaml

```diff
@@ -0,0 +1,125 @@
+      values:
+# TODO: handle per node pool availability zone limits
+{{- $azs := .Values.availability_zones }}
+{{- if index .NodePool.ConfigItems "availability_zones"}}
+        {{    $azs = split .NodePool.ConfigItems.availability_zones "," }}
+{{- end}}
+{{- range $az :=  $azs }}
+      - "{{ $az }}"
+{{- end }}
+{{- $taints := .NodePool.Taints}}
+{{- if $taints }}
+  taints:
+  {{- range $taints }}
+    {{- $taint := . }}
+      - key: {{$taint.Key}}
+        effect: {{$taint.Effect}}
+        {{- if $taint.Value }}
+        value: {{$taint.Value }}
+        {{- end }}
+  {{- end}}
+{{- end}}
+
+  startupTaints:
+  - key: zalando.org/node-not-ready
+    effect: NoSchedule
+
+  labels:
+    # these labels are normally set by kubelet on start-up, but because
+    # karpenter already creates the node object ahead of the instance start, we
+    # need to set them here as well.
+    lifecycle-status: ready
+    node.kubernetes.io/node-pool: "{{ .NodePool.Name }}"
+    node.kubernetes.io/role: worker
+    node.kubernetes.io/profile: "{{ .NodePool.Profile }}"
+    cluster-lifecycle-controller.zalan.do/replacement-strategy: none
+{{- if index .NodePool.ConfigItems "labels"}}
+  {{- range split .NodePool.ConfigItems.labels ","}}
+    {{- $label := split . "="}}
+    {{index $label 0}}: {{index $label 1}}
+  {{- end}}
+{{- end}}
+
+  # ttlSecondsAfterEmpty: 30
+  providerRef:
+    name: {{ .NodePool.Name }}-template
+---
+apiVersion: karpenter.k8s.aws/v1alpha1
+kind: AWSNodeTemplate
+metadata:
+  name: {{ .NodePool.Name }}-template
+spec:
+  instanceProfile: {{ .Cluster.ID | awsValidID }}-{{ .NodePool.Name }}-InstanceProfile
+  amiFamily: Custom
+  amiSelector:
+    environment: production
+    purpose: zalando-ubuntu-kubernetes
+    version: v1.21.14
+    revision: master-236
+  metadataOptions:
+    httpTokens: optional
+  subnetSelector:
+    Name: "dmz-eu-central-1*" # TODO: use a better tag
+  securityGroupSelector:
+    karpener.sh/discovery: "WorkerNodeSecurityGroup"
+  blockDeviceMappings:
+    - deviceName: /dev/sda1
+      ebs:
+        deleteOnTermination: {{ .NodePool.ConfigItems.ebs_root_volume_delete_on_termination}}
+        volumeSize: {{ .NodePool.ConfigItems.ebs_root_volume_size }}G
+        volumeType: gp3
+  userData: |
+{{.UserData | indent 4}}
+  tags:
+    InfrastructureComponent: "true"
+    # TODO: maybe use application label on dedicated node pools?
+    application: kubernetes
+    component: worker
+    environment: "{{ .Cluster.Environment }}"
+    Name: "{{ .NodePool.Name }} ({{ .Cluster.ID }})"
+    # TODO: are these tags still needed?
+    # Do we want a node pool tag for cost tracking reasons?
+    node.kubernetes.io/role: worker
+    node.kubernetes.io/node-pool: "{{ .NodePool.Name }}"
+# only node pools without taints should be attached to Ingress Load balancer
+{{- if or (not (index .NodePool.ConfigItems "taints")) (eq (index .NodePool.ConfigItems "taints") "dedicated=skipper-ingress:NoSchedule") }}
```

Comment:
nit: This logic should be updated to be compatible with the changes made here: https://github.com/zalando-incubator/kubernetes-on-aws/pull/5158

We can argue that it's less relevant with skipper-ingress on a dedicated node pool, but we may wan't to optimize this in the future for cost savings so IMO better to keep it flexible for now according to the referenced PR.

Reply:
Wow, what an If statement. It would be better to have it as a template helper function.
will improve it in the next PR